homeassistant/components/sensor/kwb.py

```diff
@@ -0,0 +1,101 @@
+"""
+Support for KWB Easyfire.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.kwb/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_HOST, CONF_PORT)
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pykwb==0.0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 23
+DEFAULT_TYPE = 'tcp'
+DEFAULT_RAW = 'off'
+
+MODE_SERIAL = 0
+MODE_TCP = 1
+
+CONF_TYPE = 'type'
+CONF_RAW = 'raw'
+
+SERIAL_SCHEMA = {
+    vol.Required(CONF_PORT): cv.string,
+    vol.Required(CONF_TYPE): 'serial',
+    vol.Optional(CONF_TYPE): 'raw',
+}
+
+ETHERNET_SCHEMA = {
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PORT): cv.positive_int,
+    vol.Required(CONF_TYPE): 'tcp',
+    vol.Optional(CONF_TYPE): 'raw',
+}
+
+
```

Comment:
You need to define `PLATFORM_SCHEMA` here. Right now the schemas you have defined aren't being used.

[Use the example here](https://home-assistant.io/developers/platform_example_light/) as a guide on how to get started. Since you can connect via TCP or serial, the best way is to have `CONF_HOST` and `CONF_DEVICE` be part of an exclusive group. [(Example here)](https://github.com/home-assistant/home-assistant/blob/9cad9c19f8a29c0173cb7672fd3ad95d91c34ee1/homeassistant/components/keyboard_remote.py#L39). This will make sure that only one of the two is defined, and will make it so the user doesn't need to add a redundant type config.